cmd/cloud-controller-manager/app/options/options.go

```diff
@@ -208,7 +206,7 @@ func (o *CloudControllerManagerOptions) ApplyTo(c *cloudcontrollerconfig.Config,
 		c.ClientBuilder = rootClientBuilder
 	}
 	c.VersionedClient = rootClientBuilder.ClientOrDie("shared-informers")
-	c.SharedInformers = informers.NewSharedInformerFactory(c.VersionedClient, resyncPeriod(c)())
+	c.SharedInformers = informers.NewSharedInformerFactory(c.VersionedClient, controller.DynamicResyncPeriodFunc(c.ComponentConfig.Generic.MinResyncPeriod.Duration)())
```

Comment:
This replacement is super verbose, what if you made this a method, so you could just say something like `c.ComponentConfig.Generic.DynamicResyncPeriodFunc`?

Reply:
Actually it doesn't need to return a function, so just `c.ComponentConfig.Generic.DynamicResyncPeriod` might be fine.

Reply:
Sorry, not sure if I understand correctly. `c.ComponentConfig.Generic` is a config struct in types.go that holds configuration like `MinResyncPeriod` got from CLI option, I think adding a method to it might break its style?
https://github.com/kubernetes/kubernetes/blob/58596b2bf5eb0d84128fa04d0395ddd148d96e51/pkg/controller/apis/config/types.go#L122-L130

The reason it's made to return a function is another place is using the returned closure to generate different resyncPeriod for controllers:
https://github.com/kubernetes/kubernetes/blob/58596b2bf5eb0d84128fa04d0395ddd148d96e51/cmd/kube-controller-manager/app/controllermanager.go#L329-L332

The verbose function name was trying to keep same style as the other two resyncPeriod generating functions.
https://github.com/kubernetes/kubernetes/blob/d88304507d3b3ff8a15f0948c21be5892693a12f/pkg/controller/controller_utils.go#L103-L115

Please correct me if I misunderstand you.